src/middlewared/middlewared/scripts/wait_to_hang_and_dump_core.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding=utf-8 -*-
+import logging
+import os
+import pathlib
+import re
+import socket
+import subprocess
+import time
+
+from middlewared.client import Client
+
+logger = logging.getLogger(__name__)
+
+
+def main():
+    logging.basicConfig(level=logging.INFO, format="[%(asctime)s] %(message)s")
+
+    interval = 10
+    logger.info("Probing middleware accessibility with %d seconds interval", interval)
+    while True:
+        try:
+            with Client():
+                pass
+        except socket.timeout:
+            logger.info("Caught timeout, dumping core")
+            dump_core()
+            break
+        else:
+            time.sleep(interval)
+
+
+def dump_core():
+    middlewared_pid = int(
+        re.match(
+            r"MainPID=([0-9]+)",
+            subprocess.check_output(
+                "systemctl show --property MainPID middlewared".split(),
+                encoding="utf-8",
+            ),
+        ).group(1),
+    )
+    logger.info("middlewared PID: %d", middlewared_pid)
+
+    subprocess.run(
+        ["gdb", "-p", str(middlewared_pid), "-batch", "-ex", "generate-core-file"],
+        check=True,
+    )
+
+    core_file = sorted(pathlib.Path.cwd().glob("core.*"), key=os.path.getmtime, reverse=True)[0]
```

Comment:
If another program core dumps after this is finished and before we `gzip` it then we'll get the wrong core file right? The naming of the core file should have some identifiable information in it so we get the middlewared core dump.